src/Microsoft.Sbom.Api/SbomValidator.cs

```diff
@@ -104,7 +104,8 @@ public async Task<SbomValidationResult> ValidateSbomAsync(
         await recorder.FinalizeAndLogTelemetryAsync();
 
         var errors = recorder.Errors.Select(error => error.ToEntityError()).ToList();
-        return new SbomValidationResult(!errors.Any(), errors);
+        var hasExceptions = recorder.Exceptions.Any();
+        return new SbomValidationResult(!errors.Any() && !hasExceptions, errors);
```

Comment:
This should work, but it feels a bit forced. If the root cause of the problem is a mismatch between the recorder's result and the app's result, what if we just removed the mismatch? Imagine that we changed 3 things:

First, add this in IRecorder.cs:
```
    public Result Result { get; }
```

Then add the implementation in `TelemetryRecorder.cs`:
```
    public Result Result => result;
```

Finally, replace line 107 (where we try to mirror what the `TelemetryRecorder` does) with this:
```
  return new SbomValidationResult(recorder.Result == Result.Success, errors);
```

... and you'd need to update the concretion pin